boot/freeldr/freeldr/ntldr/wlmemory.c

```diff
@@ -47,6 +47,7 @@ WinLdrInsertDescriptor(IN OUT PLOADER_PARAMETER_BLOCK LoaderBlock,
 
 /* GLOBALS ***************************************************************/
 
+ULONG_PTR AcpiTableSource;
```

Comment:
I think this could stay a `PVOID`, this would allow to remove the `ULONG_PTR` cast in this file when you call `FindAcpiTable()` (or in your original code, when you assigned `AcpiTable` to `AcpiTableSource` ), and remove the `(PVOID)` cast in winldr.c when you call the RtlCopyMemory.
This would only add one single cast in the `MempUnmapPage` call however.